bin/docker_connection.py

```diff
@@ -23,7 +23,12 @@ async def _start_docker_controller(docker_pipe: aioprocessing.AioConnection):
     this loop to be cancellable when the user desires to exit. Having any blocking I/O operation in this loop would mean
     the potential for the child process to be stuck while the user is trying to exit from Hummingbot.
     """
-    docker_client: docker.APIClient = docker.APIClient(base_url="unix://var/run/docker.sock")
+    try:
+        docker_client: docker.APIClient = docker.APIClient(base_url="unix://var/run/docker.sock")
+    except Exception as e:
+        await docker_pipe.coro_send(e)
```

Comment:
You shouldn't stop the loop here. The architecture right now is to detect the error when `docker_ipc()` or similar functions get called. If you simply plug one message into the pipe like this, all subsequent IPC calls after the first one would hang.

Reply:
Instead, you should intercept the exception and store it locally first. Then, in the loop the follows - if there's any initialization error recorded, you always send that exception object back for all requests.

Reply:
The other way you can do this is by making sure the docker side of the pipe is fully closed if the initialization step failed - in that case, you can fully exit the loop and the child process.

If you want to do that, then you need to make sure the docker pipe is closed on the master process's side just after the child process has started (you'll need to use an `aioprocessing.Event` to do this), s.t. it doesn't remain open on the master side when the child process has started.

If you do that, then any `docker_ipc()` call going into the hummingbot pipe would give you broken pipe error, and you can infer that the child process has exited. But you'll also lose the original exception object that tells why the `APIClient` initialization failed. So while it may look a tiny bit simpler architecturally (arguable - since you need to add an Event), I don't recommend it.